experimental/StandardFiniteFields/StandardFiniteFields.jl

```diff
@@ -0,0 +1,462 @@
+using Oscar
+
+IntegerUnion   = Union{Integer, ZZRingElem}
+PrimeField     = Union{Nemo.fpField, Nemo.FpField}
+PrimeFieldElem = Union{fpFieldElem, FpFieldElem}
+PrimeFieldMatrix = Union{FpMatrix, fpMatrix}
+
+# NOTE: These give missing features to OSCAR/Nemo that will likely be added in the near future.
+
+# TODO : Should be fixed in Nemo
+function (k::Nemo.FpField)(a::Vector)
+  @assert length(a) == 1
+  return k(a[1])
+end
+function (k::FqPolyRepField)(a::Vector)
+  return k(polynomial(GF(ZZ(characteristic(k))), a))
+end
+
+
+# TODO : Should be fixed in Hecke for prime fields
+function coords(x::FinFieldElem)
+    return absolute_coordinates(x)
+end
+function coords(x::PrimeFieldElem)
+    return [x]
+end
+
+# TODO : this should be pushed to Nemo.jl/src/flint/gfp_fmpz_mat.jl
+# BUG the fact that this is missing is a major performance problem
+import Base: inv
+using FLINT_jll
+const libflint = FLINT_jll.libflint
+function inv(a::FpMatrix)
+  !is_square(a) && error("Matrix must be a square matrix")
+  z = similar(a)
+  r = ccall((:fmpz_mod_mat_inv, libflint), Int,
+          (Ref{FpMatrix}, Ref{FpMatrix}), z, a)
+  !Bool(r) && error("Matrix not invertible")
+  return z
+end
+
+# TODO : this just makes sense for writing general code
+import Base: big
+big(a::ZZRingElem) = a
+
+
+# _attributes = [
+    # :is_standard_finite_field      - bool
+    # :is_standard_prime_field       - bool
+    # :primitive_powers_in_tower_basis - Matrix{FinFieldElem}
+    # :tower_basis                  - Matrix{FFieldElem} = inv(:primitive_powers_in_tower_basis)
+    # NOTE I guess we used ZZ ring elements for these instead of Int; does it matter?
+    # :steinitz_prime_degree         - Dict{Int, Dict{Int, Int}}
+    # :standard_extensions          - Dict{Int, FinField} ]
+function set_standard_prime_field!(F::PrimeField)
+    get_attribute!(F, :is_standard_prime_field) do
+        set_attribute!(F, :standard_extensions, Dict{ZZRingElem, FinField}(1 => F))
+        set_attribute!(F, :primitive_powers_in_tower_basis, identity_matrix(F, 1))
+        true
+    end
+end
+function set_standard_finite_field!(F::FinField)
+    set_attribute!(F, :is_standard_finite_field, true)
+end
+function set_primitive_powers_in_tower_basis!(F::FinField, M::PrimeFieldMatrix)
+    set_attribute!(F, :primitive_powers_in_tower_basis, M)
+    set_attribute!(F, :tower_basis, inv(M))
+end
+function set_steinitz_prime_degree!(F::FinField, r::IntegerUnion, k::IntegerUnion, nr::IntegerUnion)
+    spd = get_attribute!(F, :steinitz_prime_degree, Dict{ZZRingElem, Dict{ZZRingElem, ZZRingElem}}())
+    spdr = get!(spd, r, Dict{ZZRingElem, ZZRingElem})
+    spdr[k] = nr
+end
+function set_steinitz_prime_degree!(f::Function, F::FinField, r::IntegerUnion, k::IntegerUnion)
```

Comment:
I guess you are right. I felt like they should be there, but maybe not. It makes the most sense to always set these using `get_steinitz_prime_degree!` instead, since we always want to check if it is already computed before computing it again.